app/modules/pipe.c

```diff
@@ -104,22 +141,115 @@ static char getsize_delim (lua_State *L, int ndx, int *len) {     // [-0, +0, v]
   return delim;
 }
 
-/* Lua callable methods */
+/*
+** Read CB Initiator AND pipe_write. If arg1 == the pipe, then this is a pipe
+** write(); otherwise it is the Lua CB wapper for the task post. This botch allows
+** these two functions to share Upvals within the Lua 5.1 VM:
+*/
+#define UVpipe  lua_upvalueindex(1)  // The pipe table object
+#define UVfunc  lua_upvalueindex(2)  // The CB's Lua function
+#define UVprio  lua_upvalueindex(3)  // The task priority
+#define UVstate lua_upvalueindex(4)  // Pipe state;
+#define CB_NOT_USED       0
+#define CB_ACTIVE         1
+#define CB_WRITE_UPDATED  2
+#define CB_QUIESCENT      4
+/*
+** Note that nothing precludes the Lua CB function from itself writing to the 
+** pipe and in this case this routine will call itself recursively. 
```

Comment:
Recursively, or merely retasking itself?

Reply:
Recursively vs retasking.  I leave it to you to propose alternative wording. 

The real point here was that node.output() used to be a minefield as it was really breaking the VM architectural assumptions in that the print base function was never intended to embed a Lua CB, so doing a `print` inside this Lua CB code could trash the whole runtime environment and PANIC the processor.  

Because the output interface now uses pipes and the tasking mechanism we now comply with the architecture and there is actually nothing stopping the output CB doing a print.